src/sqs/service/SqsMessageService.ts

```
@@ -0,0 +1,107 @@
+import DecompressionService from '../../decompression/service/DecompressionService';
+
+const MAX_NUMBER_OF_MESSAGES = 10;
+
+export default class SqsMessageService {
+  private readonly sqs: AWS.SQS;
+  private readonly sqsReceiveMessageRequest: AWS.SQS.ReceiveMessageRequest;
+
+  constructor(
+    awsSecretKey: string,
+    awsAccessKeyId: string,
+    private readonly sqsQueueUrl: string,
+    region = 'us-east-1'
+  ) {
+    const sqsConfig: AWS.SQS.ClientConfiguration = {
+      region,
+      accessKeyId: awsAccessKeyId,
+      secretAccessKey: awsSecretKey
+    };
+    this.sqs = new AWS.SQS(sqsConfig);
+    this.sqsReceiveMessageRequest = {
+      QueueUrl: this.sqsQueueUrl,
+      MaxNumberOfMessages: MAX_NUMBER_OF_MESSAGES
+    };
+  }
+
+  get QueueUrl(): string {
+    return this.sqsQueueUrl;
+  }
+
+  public async getAllMessages(options?: GetMessageOptions): Promise<AWS.SQS.Message[]> {
+    let messages: AWS.SQS.Message[] = [];
+    let tryingToGetMessages = true;
+    while (tryingToGetMessages) {
+      try {
+        const messageGroup = await this.getMessageGroup();
+        if (messageGroup === undefined || messageGroup.length === 0) {
+          tryingToGetMessages = false;
+        } else {
+          messages = messages.concat(messageGroup);
+        }
+      } catch (e) {
+        if (!this.isRetryableSqsError(e)) {
+          throw e;
+        }
+      }
+    }
+    return Q.all(messages.map(message => this.finalizeMessage(message, options)));
+  }
+
+  public acknowledgeMessage(message: AWS.SQS.Message): Q.Promise<any> {
+    const deferred = Q.defer<any>();
+    if (message.ReceiptHandle === undefined) {
+      return Q.reject(new Error('Receipt Handle on message is undefined'));
+    }
+    this.sqs.deleteMessage({
+      QueueUrl: this.sqsQueueUrl,
+      ReceiptHandle: message.ReceiptHandle
+    }, (err) => {
+      if (err) {
+        deferred.reject(`Message "${message.MessageId}" not acknowledged.`);
```

You are hiding the error message from the delete message request here. Please include the err in your string here.